macOS/LocalPackages/DataBrokerProtection/Sources/DataBrokerProtection/Shared/Operations/DataBrokerProfileQueryOperationManager.swift

```diff
@@ -113,127 +114,85 @@ struct DataBrokerProfileQueryOperationManager: OperationsManager {
                                    shouldRunNextStep: @escaping () -> Bool) async throws {
         Logger.dataBrokerProtection.log("Running scan operation: \(brokerProfileQueryData.dataBroker.name, privacy: .public)")
 
-        guard let brokerId = brokerProfileQueryData.dataBroker.id, let profileQueryId = brokerProfileQueryData.profileQuery.id else {
+        // 1. Validate that the broker and profile query data objects each have an ID:
+        guard let brokerId = brokerProfileQueryData.dataBroker.id,
+              let profileQueryId = brokerProfileQueryData.profileQuery.id else {
             // Maybe send pixel?
             throw OperationsError.idsMissingForBrokerOrProfileQuery
         }
 
         defer {
             try? database.updateLastRunDate(Date(), brokerId: brokerId, profileQueryId: profileQueryId)
-            Logger.dataBrokerProtection.log("Finished scan operation: \(brokerProfileQueryData.dataBroker.name, privacy: .public)")
             notificationCenter.post(name: DataBrokerProtectionNotifications.didFinishScan, object: brokerProfileQueryData.dataBroker.name)
+            Logger.dataBrokerProtection.log("Finished scan operation: \(brokerProfileQueryData.dataBroker.name, privacy: .public)")
         }
 
+        // 2. Set up dependencies used to report the status of the scan job:
         let eventPixels = DataBrokerProtectionEventPixels(database: database, handler: pixelHandler)
```

Comment:
Not directly relevant to this PR, but I think part of the problem is some of these classes have confusing names. E.g. EventPixels is not remotely what I would have expected from the name

StageDurationCalculator too I think is a bit weird, since that describes one of the things it does to send the pixels it actually cares about. I.e. it's not named after the the reason it exists/utility it actually provides. We don't seem to use the stage duration from this anywhere else. And what's worse, in at least one place we seem to calculate it again separately